lib/nats/io/client.rb

```diff
@@ -1484,11 +1470,6 @@ def attempt_reconnect
       begin
         srv = select_next_server
 
-        # Establish TCP connection with new server
-        @io = create_socket
```

Comment:
Here also moved hostname assignment to be before socket (and tls context) creation, see above